transformations/tests/test_scc_cuf.py

```diff
@@ -0,0 +1,394 @@
+    assert horizontal.index not in loop_variables
+    assert vertical.index in loop_variables
+    argument_arrays = [arg for arg in routine.arguments if isinstance(arg, sym.Array)]
+    for argument_array in argument_arrays:
+        dims = FindVariables().visit(argument_array.dimensions)
+        assert blocking.index in dims or blocking.size in dims
+    # TODO: assert for local arrays!
+    # arrays = [var for var in routine.variables if isinstance(var, sym.Array)]
+
+
+def check_subroutine_kernel(routine, horizontal, vertical, blocking):
+    _check_subroutine_kernel(routine=routine, horizontal=horizontal, vertical=vertical, blocking=blocking)
+    assert "ATTRIBUTES(GLOBAL)" in routine.prefix
+    assignments = FindNodes(Assignment).visit(routine.body)
+    assert "THREADIDX%X" in [_.rhs for _ in assignments]
+    assert "BLOCKIDX%Z" in [_.rhs for _ in assignments]
+
+
+def check_subroutine_device(routine, horizontal, vertical, blocking):
+    _check_subroutine_kernel(routine=routine, horizontal=horizontal, vertical=vertical, blocking=blocking)
+    assert "ATTRIBUTES(DEVICE)" in routine.prefix
+    assert horizontal.index in routine.arguments
+    assert blocking.index in routine.arguments
+
+
+def check_subroutine_elemental_device(routine):
+    assert "ATTRIBUTES(DEVICE)" in routine.prefix
+    assert "ELEMENTAL" not in routine.prefix
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_scc_cuf_simple(frontend, horizontal, vertical, blocking):
+
+    fcode_driver = """
+  SUBROUTINE driver(nlon, nz, nb, tot, q, t, z)
+    INTEGER, INTENT(IN)   :: nlon, nz, nb  ! Size of the horizontal and vertical
+    INTEGER, INTENT(IN)   :: tot
+    REAL, INTENT(INOUT)   :: t(nlon,nz,nb)
+    REAL, INTENT(INOUT)   :: q(nlon,nz,nb)
+    REAL, INTENT(INOUT)   :: z(nlon,nz+1,nb)
+    INTEGER :: b, start, end, ibl, icend
+
+    start = 1
+    end = tot
+    do b=1,end,nlon
+      ibl = (b-1)/nlon+1
+      icend = MIN(nlon,tot-b+1)
+      call kernel(start, icend, nlon, nz, q(:,:,b), t(:,:,b), z(:,:,b))
+    end do
+  END SUBROUTINE driver
+"""
+
+    fcode_kernel = """
+  SUBROUTINE kernel(start, end, nlon, nz, q, t, z)
+    INTEGER, INTENT(IN) :: start, end  ! Iteration indices
+    INTEGER, INTENT(IN) :: nlon, nz    ! Size of the horizontal and vertical
+    REAL, INTENT(INOUT) :: t(nlon,nz)
+    REAL, INTENT(INOUT) :: q(nlon,nz)
+    REAL, INTENT(INOUT) :: z(nlon,nz)
+    INTEGER :: jl, jk
+    REAL :: c
+
+    c = 5.345
+    DO jk = 2, nz
+      DO jl = start, end
+        t(jl, jk) = c * k
+        q(jl, jk) = q(jl, jk-1) + t(jl, jk) * c
+      END DO
+    END DO
+    
+    DO jk = 2, nz
+      DO jl = start, end
+        z(jl, jk) = 0.0 
+      END DO
+    END DO
+    
+    ! DO JL = START, END
+    !   Q(JL, NZ) = Q(JL, NZ) * C
+    ! END DO
+  END SUBROUTINE kernel
+"""
+    kernel = Subroutine.from_source(fcode_kernel, frontend=frontend)
+    driver = Subroutine.from_source(fcode_driver, frontend=frontend)
```

Comment:
If you provide here `definitions=[kernel]` you can avoid the `enrich_calls` call